src/fluree/db/json_ld/transact.cljc

```diff
@@ -188,22 +192,33 @@
                          ;; means we cannot transact shacl in same txn as
                          ;;data and have it enforced.
                          (<? (get-subject-types db-before sid new-type-sids)))
-          class-shapes (<? (shacl/class-shapes db-before classes))
+
+          class-shapes   (<? (shacl/class-shapes db-before classes))
           referring-pids (when shacl-target-objects-of?
-                           (cond-> (map flake/p (<? (query-range/index-range db-before :opst = [sid])))
+                           (cond-> (<? (query-range/index-range db-before :opst = [sid] {:flake-xf (map flake/p)}))
```

Comment:
This PR reminded me that we can pass these transducers in to `index-range` 😄 I'm glad to see that getting used more